src/SignalR/common/Http.Connections/src/Internal/HttpConnectionContext.cs

```diff
@@ -313,6 +331,9 @@ private async Task WaitOnTasks(Task applicationTask, Task transportTask, bool cl
                     Application?.Output.Complete(transportTask.Exception?.InnerException);
                     Application?.Input.Complete();
 
+                    // Trigger ConnectionClosed
+                    _connectionClosedTokenSource.Cancel();
```

Comment:
We should dispatch to the thread pool before calling Cancel().

Reply:
Ah true, good catch